src/Compilers/VisualBasic/Portable/Symbols/Attributes/SourceAttributeData.vb

```diff
@@ -233,6 +237,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                             foundMatch = parameterType = lazySystemType
                             k += 1
 
+                        Case CByte(SignatureTypeCode.SZArray)
+                            ' skip over and check the next byte
+                            foundMatch = parameterType.IsArrayType
+
                         Case Else
                             Return -1
                     End Select
```

Comment:
The method has some symbol casts and calls some extension methods that would need to be unified. I think having two parallel implementations is ok here. They should not change much in the future.

---
In reply to: [80366315](https://github.com/dotnet/roslyn/pull/14023#discussion_r80366315) [](ancestors = 80366315)